packages/x-charts/src/ChartsLegend/PiecewiseColorLegend.tsx

```diff
@@ -1,132 +1,263 @@
-        return null;
-      }
-
-      const data = {
-        ...(isFirst
-          ? { min: null, formattedMin: null }
-          : { min: colorMap.thresholds[index - 1], formattedMin: formattedLabels[index - 1] }),
-        ...(isLast
-          ? { max: null, formattedMax: null }
-          : { max: colorMap.thresholds[index], formattedMax: formattedLabels[index] }),
-      };
-
-      const label = labelFormatter(data);
-
-      if (label === null) {
-        return null;
-      }
-
-      return {
-        id: label,
-        color,
-        label,
-        minValue: data.min,
-        maxValue: data.max,
-      };
-    })
-    .filter(notNull);
-
-  return (
-    <LegendPerItem
-      {...other}
-      itemsToDisplay={itemsToDisplay}
-      onItemClick={
-        onItemClick
-          ? (event, i) => onItemClick(event, piecewiseColorContextBuilder(itemsToDisplay[i]), i)
-          : undefined
-      }
-    />
-  );
-}
+const RootElement = styled('ul', {
+  name: 'MuiPiecewiseColorLegend',
+  slot: 'Root',
+  overridesResolver: (props, styles) => styles.root,
+})<{ ownerState: PiecewiseColorLegendProps }>(({ theme, ownerState }) => {
+  return {
+    ...theme.typography.caption,
+    color: (theme.vars || theme).palette.text.primary,
+    lineHeight: undefined,
+    display: 'flex',
+    flexDirection: ownerState.direction === 'vertical' ? 'column' : 'row',
+    flexShrink: 0,
+    gap: theme.spacing(0.5),
+    listStyleType: 'none',
+    paddingInlineStart: 0,
+    marginBlock: theme.spacing(1),
+    marginInline: theme.spacing(1),
+    width: 'max-content',
+    [`button.${piecewiseColorLegendClasses.item}`]: {
+      // Reset button styles
+      background: 'none',
+      border: 'none',
+      padding: 0,
+      cursor: ownerState.onItemClick ? 'pointer' : 'unset',
+      fontFamily: 'inherit',
+      fontWeight: 'inherit',
+      fontSize: 'inherit',
+      letterSpacing: 'inherit',
+      color: 'inherit',
+    },
+    [`.${piecewiseColorLegendClasses.item}`]: {
+      display: 'flex',
+      gap: theme.spacing(0.5),
+    },
+    [`li :not(.${piecewiseColorLegendClasses.minLabel}, .${piecewiseColorLegendClasses.maxLabel}) .${piecewiseColorLegendClasses?.mark}`]:
+      {
+        alignSelf: 'center',
+      },
+    [`&.${piecewiseColorLegendClasses.horizontal}`]: {
+      alignItems: 'center',
+      [`.${piecewiseColorLegendClasses.item}`]: {
+        flexDirection: 'column',
+      },
+      [`&.${piecewiseColorLegendClasses.below}, &.${piecewiseColorLegendClasses.left}`]: {
+        alignItems: 'start',
+      },
+      [`&.${piecewiseColorLegendClasses.above}, &.${piecewiseColorLegendClasses.right}`]: {
+        alignItems: 'end',
+      },
+      [`.${piecewiseColorLegendClasses.minLabel}`]: {
+        alignItems: 'end',
+      },
+      [`&.${piecewiseColorLegendClasses.extremes}`]: {
+        [`.${piecewiseColorLegendClasses.minLabel}, .${piecewiseColorLegendClasses.maxLabel}`]: {
```

Comment:
Is it necessary to have both extremes and min/max label? or could we reduce specificity?

Reply:
Yeah, the `.extremes` is added to the parent. The `min/max` target the label themselves